indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamSamplerSpec.java

```diff
@@ -92,9 +93,22 @@ public SamplerResponse sample()
       );
       inputFormat = null;
     } else {
+      RecordSupplier<PartitionIdType, SequenceOffsetType, RecordType> recordSupplier;
+
+      try {
+        recordSupplier = createRecordSupplier();
+      }
+      catch (Exception e) {
+        if (e.getCause() != null) {
+          throw new SamplerException(e, "Unable to create RecordSupplier: %s. Cause: %s", e.getMessage(), e.getCause().getMessage());
```

Comment:
How about using this Druid style, as follows:

```suggestion
          throw new SamplerException(e, "Unable to create RecordSupplier: %s because: %s", e.getMessage(), e.getCause().getMessage());
```

Reply:
It's very useful to print the cause, but in this situation, I might want to know the root cause, so maybe we can use `Throwables.getRootCause(xxx)` instead.